applications/launcher/eventfilter.cpp

```diff
@@ -79,58 +99,93 @@ int parentCount(QQuickItem* obj){
+                        mouseEvent->type(), mouseEvent->localPos(), mouseEvent->windowPos(),
+                        mouseEvent->screenPos(), mouseEvent->button(), mouseEvent->buttons(),
+                        mouseEvent->modifiers()
+                        );
             auto widgetPos = globalPos((QQuickItem*)postWidget);
             auto localPos = event->localPos();
             localPos.setX(pos.x() - widgetPos.x());
             localPos.setY((pos.y()) - widgetPos.y());
             event->setLocalPos(localPos);
+            qDebug() << "postEvent: " << event;
             QGuiApplication::postEvent(postWidget, event);
         }
     }
 }
 
+bool isTabletEvent(QEvent::Type type) {
+    return type == QEvent::TabletPress ||
+            type == QEvent::TabletRelease ||
+            type == QEvent::TabletMove;
+}
+
+bool isTouchEvent(QEvent::Type type) {
+    return type == QEvent::TouchBegin ||
+            type == QEvent::TouchUpdate ||
+            type == QEvent::TouchEnd;
+}
+
+QEvent::Type tabletEventTypeToMouseEventType(QEvent::Type type) {
+    switch(type) {
+    case QEvent::TabletPress:
+        return QEvent::MouseButtonPress;
+    case QEvent::TabletRelease:
+        return QEvent::MouseButtonRelease;
+    case QEvent::TabletMove:
+        return QEvent::MouseMove;
+    default:
+        return type;
+    }
+    return type;
+}
+
+QEvent::Type touchEventTypeToMouseEventType(QEvent::Type type) {
+    switch(type) {
+    case QEvent::TouchBegin:
+        return QEvent::MouseButtonPress;
+    case QEvent::TouchEnd:
+        return QEvent::MouseButtonRelease;
+    case QEvent::TouchUpdate:
+        return QEvent::MouseMove;
+    default:
+        return type;
+    }
+    return type;
+}
+
 bool EventFilter::eventFilter(QObject* obj, QEvent* ev){
     auto type = ev->type();
     bool filtered = QObject::eventFilter(obj, ev);
     if(!filtered){
-        if(type == QEvent::TabletPress){
-#ifdef DEBUG_EVENTS
-            qDebug() << ev;
-#endif
-            postEvent(QMouseEvent::MouseButtonPress, ev, root);
-        }else if(type == QEvent::TabletRelease){
-#ifdef DEBUG_EVENTS
-            qDebug() << ev;
-#endif
-            postEvent(QMouseEvent::MouseButtonRelease, ev, root);
-        }else if(type == QEvent::TabletMove){
-#ifdef DEBUG_EVENTS
-            qDebug() << ev;
-#endif
-            postEvent(QMouseEvent::MouseMove, ev, root);
+        if (isTabletEvent(type) || isTouchEvent(type)) {
+            QMouseEvent* mouseEvent = nullptr;
+            if (isTabletEvent(type)) {
+                mouseEvent = tabletToMouseEvent(tabletEventTypeToMouseEventType(type), ev);
+            }
+            else if (Settings::instance().getDeviceType() == DeviceType::RM2 && isTouchEvent(type)) {
+                mouseEvent = touchToMouseEvent(touchEventTypeToMouseEventType(type), ev);
+            }
+            if (mouseEvent) {
+                postEvent(mouseEvent, root);
+                delete mouseEvent;
+            }
```

Comment:
```suggestion
            if (isTabletEvent(type)) {
                postEvent(
                    tabletToMouseEvent(tabletEventTypeToMouseEventType(type), ev),
                    root
                );
            }
            else if (Settings::instance().getDeviceType() == DeviceType::RM2 && isTouchEvent(type)) {
                postEvent(
                    touchToMouseEvent(touchEventTypeToMouseEventType(type), ev),
                    root
                );
            }
```

Reply:
Why did you remove delete?

Reply:
I removed the entire `mouseEvent` variable.